crates/fetch/src/lib.rs

```diff
@@ -0,0 +1,313 @@
+//! A library that wraps the HTTP *fetch* API.
+#![warn(missing_docs)]
+
+mod headers;
+
+use js_sys::{ArrayBuffer, Promise, Uint8Array};
+use std::{
+    cell::{RefCell, RefMut},
+    ops::Deref,
+};
+use wasm_bindgen::{JsCast, JsValue, UnwrapThrowExt};
+use wasm_bindgen_futures::JsFuture;
+
+pub use crate::headers::Headers;
+#[doc(inline)]
+pub use web_sys::{
+    ReferrerPolicy, RequestCache, RequestCredentials, RequestMode, RequestRedirect, ResponseType,
+};
+
+/// A wrapper round `web_sys::Request`: an http request to be used with the `fetch` API.
+pub struct Request {
+    init: RefCell<web_sys::RequestInit>,
+    headers: RefCell<Option<Headers>>,
+}
+
+impl Request {
+    /// Creates a new request that will be sent to `url`.
+    ///
+    /// Uses `GET` by default. `url` can be a `String`, a `&str`, or a `Cow<'a, str>`.
+    pub fn new() -> Self {
+        Self {
+            init: RefCell::new(web_sys::RequestInit::new()),
+            headers: RefCell::new(None),
+        }
+    }
+
+    /// The request method, e.g., GET, POST.
+    ///
+    /// Note that the Origin header is not set on Fetch requests with a method of HEAD or GET.
+    pub fn method(&mut self, method: &str) -> &mut Self {
+        self.init.borrow_mut().method(method);
+        self
+    }
+
+    /// Get access to the headers object for this request.
+    ///
+    /// If you are going to insert new headers, note that
+    /// [some names are forbidden](https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_header_name)
+    /// and if they are set then sending the request will error.
```

Comment:
Need to check the spec to see what happens, it's possible the offending headers are silently discarded.